src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/ProcessorClient.cs

```diff
@@ -77,22 +80,26 @@ internal class ProcessorClient(
                     url,
                     request,
                     cancellationToken: cancellationToken),
+            additionalResponseValidator,
             cancellationToken);
     }
 
     private async Task<Either<ActionResult, Unit>> SendDelete(
         string url,
+        Func<HttpResponseMessage, ActionResult?>? additionalResponseValidator = null,
         CancellationToken cancellationToken = default)
     {
         return await SendRequest<Unit>(
             () => httpClient.DeleteAsync(
                 url,
                 cancellationToken: cancellationToken),
+            additionalResponseValidator,
             cancellationToken);
     }
 
     private async Task<Either<ActionResult, TResponse>> SendRequest<TResponse>(
         Func<Task<HttpResponseMessage>> requestFunction,
+        Func<HttpResponseMessage, ActionResult?>? additionalResponseValidator = null,
```

Comment:
I'd potentially rename this to `additionalFailedResponseValidator`, since it is only used in non-success cases. But other than that, I think this looks good :)

Reply:
I've renamed to `customResponseHandler` to keep it open-ended and moved it so that it can potentially handle any and all responses, failures or otherwise.  Doesn't have to be solely for validation anymore but could be used to do something custom with success cases as well.